### PR TITLE
Update Codecov ghaction

### PR DESCRIPTION
**Pull request by Stoops-ML**
New test.

### CODE REVIEW

**Comment by Stoops-ML** [comment]
Codecov is now running as expected. A base report is missing which is why there's no comparison in the Codecov report comment. This should be resolved when this PR is merged (as there'll be an uploaded coverage report on this commit).

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -72,8 +72,23 @@ jobs:
...
+      - uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v4
+      - name: Generate coverage report
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install pytest
+          python -m pip install pytest-cov
+          python -m pip install .[dev]
+          python -m pytest --cov=./ --cov-report=xml:pytest_cov.xml
```
Comment (astrojuanlu) [inline .github/workflows/workflow.yml]: Can this be 2 actions maybe?

```suggestion
      - name: Install dependencies
        run: python -m pip install .[dev]
      - name: Generate coverage report
        run: python -m pytest --cov=./ --cov-report=xml:pytest_cov.xml
```

(I'm guessing that `.[dev]` already has `pytest` and `pytest-cov` but please check)

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -72,8 +72,23 @@ jobs:
...
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v4
+        with:
+          directory: ./coverage/reports/
+          fail_ci_if_error: true
+          files: ./coverage.xml,./pytest_cov.xml,!./cache
+          flags: unittests
+          name: codecov-umbrella
+          token: ${{ secrets.CODECOV_TOKEN }}
+          verbose: true
```
Comment (astrojuanlu) [inline .github/workflows/workflow.yml]: Oops empty line at the end

File: .github/workflows/workflow.yml (hunk 1 above)
Comment (Stoops-ML) [reply .github/workflows/workflow.yml:84]: Can you add `python -m pip install --upgrade pip` before installing `.[dev]` please

File: .github/workflows/workflow.yml (hunk 2 above)
Comment (Stoops-ML) [reply .github/workflows/workflow.yml]: Completed in [67262c9](https://github.com/poliastro/czml3/pull/131/commits/67262c992206ef8d82744b25e9ec5a7227f49b0f)

File: .github/workflows/workflow.yml (hunk 1 above)
Comment (Stoops-ML) [reply .github/workflows/workflow.yml]: Applied suggestion with the addition of upgrading `pip` in [2ddc2af](https://github.com/poliastro/czml3/pull/131/commits/2ddc2af82ff5c9b965b29c8c7c6e8eaa66282ecf)

**Review by astrojuanlu — APPROVED** [review]

**Pull request merged**